src/ucp/am/eager_multi.c

```diff
@@ -145,8 +145,11 @@ static UCS_F_ALWAYS_INLINE ucs_status_t ucp_am_eager_multi_bcopy_send_func(
         packed_size = uct_ep_am_bcopy(uct_ep, UCP_AM_ID_AM_FIRST,
                                       ucp_am_eager_multi_bcopy_pack_args_first,
                                       &pack_ctx, 0);
-        status      = ucp_am_handle_user_header_send_status(
-                req, ucp_am_multi_bcopy_send_func_status(packed_size));
+        status      = ucp_am_multi_bcopy_send_func_status(packed_size);
+        status      = ucp_am_handle_user_header_send_status(req, status);
+        if (ucs_likely(status != UCS_ERR_NO_RESOURCE)) {
+            ucp_am_release_user_header(req);
```

Comment:
What will happen if we send msg using bcopy multi, the first fragment is sent and then msg request is aborted? I think "ucp_am_release_user_header" will be called twice in this case.

Need to make sure "ucp_am_release_user_header" is not called twice for same msg request because we are not checking if we already released the user header. The alternative is to add some check to "ucp_am_release_user_header" before releasing the header (Check if it's NULL maybe?).

Reply:
`bcopy_multi` protocol aborts request by `ucp_request_complete_send`, so it won't be called twice. In case of failure (or UCS_OK), the header will be released here.

Reply:
but if `status == UCS_ERR_NO_RESOURCE` end then request will be aborted, there is a leak

Reply:
IMO better do ucp_am_release_user_header inside ucp_am_handle_user_header_send_status
we added this helper function so all relevant action would be inside it and no code duplication..

Reply:
The problem here is that it's might be late - request is completed and released by completion callback.

Reply:
per offline discussion - need to unify all protocols to release header desc from completion callbacks and abort to avoid extra check right after send